src/packagedcode/phpcomposer.py

```diff
@@ -0,0 +1,403 @@
+        # may be we should
+        if any(h in host for h in ['github', 'bitbucket', 'gitlab']):
+            return 'https://%(host)s/%(repo)s' % locals()
+        else:
+            return repo_url
+
+    if repo_url.startswith('gist:'):
+        return repo_url
+
+    elif repo_url.startswith(('bitbucket:', 'gitlab:', 'github:')):
+        hoster_urls = {
+            'bitbucket:': 'https://bitbucket.org/%(repo)s',
+            'github:': 'https://github.com/%(repo)s',
+            'gitlab:': 'https://gitlab.com/%(repo)s',
+        }
+        hoster, repo = repo_url.split(':', 1)
+        return hoster_urls[hoster] % locals()
+    elif len(repo_url.split('/')) == 2:
+        # implicit github
+        return 'https://github.com/%(repo_url)s' % locals()
+    return repo_url
+
+
+def deps_mapper(deps, package, field_name):
+    """
+    Handle deps such as dependencies, devDependencies
+    return a tuple of (dep type, list of deps)
+    https://getcomposer.org/doc/04-schema.md#package-links
+    """
+    dep_types = {
+        'dependencies': models.dep_runtime,
+        'devDependencies': models.dep_dev,
+    }
+    resolved_type = dep_types[field_name]
+    dependencies = []
+    for name, version_constraint in deps.items():
+        dep = models.Dependency(name=name, version_constraint=version_constraint)
+        dependencies.append(dep)
+    if resolved_type in package.dependencies:
+        package.dependencies[resolved_type].extend(dependencies)
+    else:
+        package.dependencies[resolved_type] = dependencies
+    return package
+
+
+dependencies_mapper = partial(deps_mapper, field_name='dependencies')
+dev_dependencies_mapper = partial(deps_mapper, field_name='devDependencies')
+
+
+person_parser = re.compile(
+    r'^(?P<name>[^\(<]+)'
+    r'\s?'
+    r'(?P<email><([^>]+)>)?'
+    r'\s?'
+    r'(?P<url>\([^\)]+\))?$'
+).match
+
+
+def parse_person(persons):
+    """
+    https://getcomposer.org/doc/04-schema.md#authors
+    A "person" is an object with a "name" field and optionally "url" and "email".
+
+    Yield  a name, email, url tuple for a person object
+    A person can be in the form:
+        "authors": [
+            {
+                "name": "Nils Adermann",
+                "email": "naderman@naderman.de",
+                "homepage": "http://www.naderman.de",
+                "role": "Developer"
+            },
+            {
+                "name": "Jordi Boggiano",
+                "email": "j.boggiano@seld.be",
+                "homepage": "http://seld.be",
+                "role": "Developer"
+            }
+        ]
+
+    Both forms are equivalent.
+    """
+    if isinstance(persons, list):
+        for person in persons:
+            # ensure we have our three values
+            name = person.get('name')
+            email = person.get('email')
+            url = person.get('homepage')
+            yield name and name.strip(), email and email.strip('<> '), url and url.strip('() ')
+    else:
```

Comment:
Is this a good idea to exit so early  if some person cannot be parsed/made sense of?